riotfile.py

```diff
@@ -1,9 +1,52 @@
 # type: ignore
+import json
+import logging
+import os
+import sys
 from typing import List
 from typing import Tuple
+from urllib.error import HTTPError
+from urllib.request import urlopen
 
 from riot import Venv
-from riot import latest
+from riot import latest as latest_riot
+
+
+logger = logging.getLogger(__name__)
+latest = object()  # sentinel value
+
+
+def find_workflow(workflow_id):
+    """Use CircleCI API to retrieve current workflow name"""
+    try:
+        url = f"https://circleci.com/api/v2/workflow/{workflow_id}"
+        res = urlopen(url)
+        body = res.read().decode()
+        return json.loads(body)["name"]
+    except HTTPError:
+        logger.error("Error loading workflow information from the CircleC: %s", url)
+        raise
+
+
+# Set up PY_Latest to True if the current workflow is test_latest
+PY_Latest = False
+if "CIRCLE_WORKFLOW_ID" not in os.environ:
+    if "DD_USE_LATEST_VERSION" in os.environ:
+        PY_Latest = True
+    else:
+        logger.warning("not in CircleCI. Use fixed dependencies versions.")
+else:
+    PY_Latest = find_workflow(os.environ["CIRCLE_WORKFLOW_ID"]) == "test_latest"
```

Comment:
Would it be possible to avoid using the circleci api here. Instead we set an environment variable to check if we should test with latest versions. I think this might be easier to maintain.

Ex:
```suggestion
    PY_Latest =  os.getenv("DD_USE_LATEST_VERSION", 'False').lower() in ('true', '1', 't') 
```

Then we can set `DD_USE_LATEST_VERSION=True` in [test_latest](https://github.com/DataDog/dd-trace-py/blob/209be4f97c733343877e53bff274c7f1fc6db2f3/.circleci/config.yml#L1320).

Reply:
As far as I've seen in [the circleci documentation](https://circleci.com/docs/set-environment-variable/), it's possible to set env vars at job level or step level, but not at workflow level

Reply:
That's right 😞. Looking at the [circleci docs](https://circleci.com/docs/configuration-reference/#workflows) we can't set an environment variable at workflow level. The best we can do is set it in on every job and then override it in the workflow. The current approach seems like cleanest way.

Reply:
I confirm, this is a missing feature of CircleCI for a long time. It wouldn't be hard to even have the name of the workflow set in an environment variable by CircleCI like it does for the job name. Some people required that for years. If at any time this feature is added to CircleCI, it will be easy to switch to that variable directly.

Reply:
Comments added